src/Spice86.Core/Emulator/CPU/InstructionsImpl/Instructions16.cs

```diff
@@ -526,6 +528,30 @@ public class Instructions16 : Instructions16Or32 {
         Cpu.Out16(State.DX, State.AX);
     }
 
+    public override void Enter() {
+        ushort bytes = Cpu.NextUint16();
+        byte level = Cpu.NextUint8();
+        level &= 0x1F;
+        ushort sp = State.SP;
+        ushort bp = State.BP;
+        sp -= 2;
+        Memory.UInt16[MemoryUtils.ToPhysicalAddress((ushort)(State.SS + sp), 0)] = State.BP;
```

Comment:
This looks strange.
Did you mean:
```suggestion
        Memory.UInt16[(ushort)(State.SS + sp), 0] = State.BP;
```
?

Reply:
Good catch. This is fixed.